### ISSUE TITLE
How to reproduce the index building and searching using FAISS with parameters recommended by AUTOFAISS?

### ISSUE THREAD

**Issue by weitianhan** [criteo/autofaiss#212]
After building an index for my vector dataset, I get the .index file and a "index_infos.json" with contents like:
```json
{
"index_key": "OPQ128_640,IVF65536_HNSW32,PQ128x8", 
"index_param": "nprobe=1212,efSearch=2424,ht=2048", 
"index_path": "/index_path", 
"size in bytes": 1548096048, 
"avg_search_speed_ms": 100.14374415040947, 
"99p_search_speed_ms": 170.2676121937112, 
"reconstruction error %": 11.134441941976547, 
"nb vectors": 10000000, 
"vectors dimension": 512, 
"compression ratio": 13.229153337390343
}
```

My questions are:
1. How to fill in the search parameters using base FAISS, especially what is "ht" in the index_param?

I am asking it because I need to search with ID filtering. Currently I am doing it like this
```python
sel_batch = faiss.IDSelectorBatch(filter_ids)
params = faiss.IVFPQSearchParameters(
    nprobe=1212,
    quantizer_params=faiss.SearchParametersHNSW(efSearch=2424),
    sel=sel_batch
)

distances, indices = faiss_index.search(vector, k, params=params)
```
It seems working but I am not sure whether it is the correct way to do it.

2. How to build and train the index only, without actually inserting the data in the index?

I am asking it because I have my separate training set and test set. I want to train the index (e.g. clustering centroids) with my training set and then insert my test set to the index to evaluate the retrieval performance of the index.

Currently I am doing it in a tricky way which is to remove all the vectors in the index created by AUTOFAISS with training set. And then add the test set data back to this empty index (hopefully the meta structure of the index is preserved).
```python
index = faiss.read_index("trained_index_with_training_set.index")
index.reset()
index.add(test_set)
...
```

Again, it seems working but I am not sure whether it is the correct way to do it.

Please help!
